packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1068,9 +1134,42 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this._summarizer;
     }
 
-    private get summariesDisabled(): boolean {
-        return this.runtimeOptions.summaryOptions.disableSummaries === true ||
-            this.runtimeOptions.summaryOptions.summaryConfigOverrides?.disableSummaries === true;
+    public get summariesDisabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.disableSummaries === true) {
+            return true;
+        }
+        return this.summaryConfiguration.state === "disabled";
+    }
+
+    public get heuristicsDisabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.summarizerOptions?.disableHeuristics === true) {
+            return true;
+        }
+        return this.summaryConfiguration.state === "disableHeuristics";
+    }
+
+    public get summarizerClientElectionEnabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.summarizerClientElection === true) {
```

Comment:
"Fluid.ContainerRuntime.summarizerClientElection" takes precedence over anything else. If that is not present, we should look at configs.

Reply:
That is what we are doing, isn't it ?